src/pybamm/experiment/step/base_step.py

```diff
@@ -492,6 +490,10 @@ def set_up(self, new_model, new_parameter_values):
 }
 
 
+def get_unit_from(a_string: str) -> str:
+    return a_string.lstrip("0123456789.-eE ")
```

Comment:
Adding "eE" is what  patches the bug, the function was split to make it easier to test